### PR TITLE
ECIL-285 Add web-internal to local docker-compose.yml.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Run two versions of ICMS to better reflect the future production infrastructure.
Added so that our end to end tests will run on different servers and test the dynamic urls.

### CODE REVIEW

File: config/settings_test.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 FILE_UPLOAD_HANDLERS = ("web.tests.file_upload_handler.DummyFileUploadHandler",)  # type: ignore[assignment]
 APP_ENV = "test"
 
+# Ensure all endpoints are available when testing.
+IS_PRIVATE_APP = True
```
Comment (MattHolmes123) [inline config/settings_test.py:20]: TODO: At at least one unittest that changes this value and reloads the URL config.

**Review by chris-pettinga — APPROVED** [review]

File: web/tests/test_urls.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def test_ilb_admin_client_cant_access_private_urls(ilb_admin_client):
+    reload_urlconf()
+    reload_urlconf("web.urls")
+    clear_url_caches()
+
+    # Check we can't use reverse.
+    with pytest.raises(NoReverseMatch):
+        reverse("signature-list")
+
+    response = ilb_admin_client.get("/signature/")
```
Comment (marcuspp) [inline web/tests/test_urls.py:31]: This is should use reverse not the hardcoded path

File: web/menu/menu.py [hunk 3]
```diff
@@ -41,11 +41,16 @@ def has_view_access(
...
+                view_url = reverse(view_name)
+                view = resolve(view_url).func.view_class()
+                view.request = request
 
-            has_perm = view.has_permission()
+                has_perm = view.has_permission()
+
+            # Now some urls are only available on private app this code can throw exceptions.
+            except NoReverseMatch:
+                has_perm = False
```
Comment (marcuspp) [inline web/menu/menu.py:53]: The coverage report says this is not being hit.

**Review by marcuspp — APPROVED** [review]
Nice one for finding a way to test the public / private urls :+1:

File: web/menu/menu.py (hunk 3 above)
Comment (MattHolmes123) [reply web/menu/menu.py:53]: Yes that's because all tests (apart from the one I added) run with all urls enabled.

**Pull request merged**